astropy/io/fits/header.py

```diff
@@ -1694,11 +1694,11 @@ def _keyword_from_index(self, idx):
         """
 
         if idx < 0:
-            idx += len(self._cards) - 1
+            idx += len(self._cards)
```

Comment:
This change seems right, though it's bizarre that nobody noticed this before as it is there since a long time (it comes from a Pyfits 3.2 merge, 3 years ago!). So this clearly needs a test ;-)